core/statedb/statedb.go

```diff
@@ -437,44 +437,92 @@ func (s *StateDB) PrepareNft(nftIndex int64) error {
 	return nil
 }
 
-func (s *StateDB) UpdateAccountTree(accounts []int64, assets []int64) error {
-	for _, accountIndex := range accounts {
-		for _, assetId := range assets {
-			assetLeaf, err := tree.ComputeAccountAssetLeafHash(
-				s.AccountMap[accountIndex].AssetInfo[assetId].Balance.String(),
-				s.AccountMap[accountIndex].AssetInfo[assetId].LpAmount.String(),
-				s.AccountMap[accountIndex].AssetInfo[assetId].OfferCanceledOrFinalized.String(),
-			)
-			if err != nil {
-				return fmt.Errorf("compute new account asset leaf failed: %v", err)
-			}
-			err = s.AccountAssetTrees[accountIndex].Set(uint64(assetId), assetLeaf)
-			if err != nil {
-				return fmt.Errorf("update asset tree failed: %v", err)
+func (s *StateDB) IntermediateRoot(cleanDirty bool) error {
```

Comment:
Why need `cleanDirty`? seems we only use it as false.

Reply:
When we want to compute a state root after a transaction, we can set cleanDirty to true. Because stateCache itself is block level.